Request to add Traxmate Webhook for TTI Integration

FieldAppEngineerCombain opened the pull request:
<!--
Thanks for submitting a pull request. Please fill the template below,
otherwise we will not be able to process this pull request.
-->

#### Summary
<!--
A short summary, referencing related issues:
Closes #0000, References #0000, etc.
-->
- Creating a webhook template as instructed. (I hope).
- Defining token to post to Traxmate capture end-point to visualize devices on Traxmate.
- Defining path and required DevEUI to map forwarded message to device and customer on Traxmate. 

#### Changes
<!-- What are the changes made in this pull request? -->
- Adding Webhook template for Traxmate


#### Notes for Reviewers
Sorry for the mess :D
<!--
NOTE: This section is optional.

Motivate briefly why it is implemented this way, if that deviates from the
implementation proposal in the referenced issues.
- How should your reviewers approach this pull request?
- @mention reviewers with special requests or questions for them
-->

...

#### Checklist
<!-- Make sure that this pull request is complete. -->

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible, they don't break existing deployments.
- [ ] Testing: The changes are tested.
- [ ] Documentation: Relevant documentation is added or updated.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/TheThingsNetwork/lorawan-webhook-templates?pullRequest=90) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/TheThingsNetwork/lorawan-webhook-templates?pullRequest=90) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/TheThingsNetwork/lorawan-webhook-templates?pullRequest=90) it.</sub>

nejraselimovic reviewed (changes requested):
Hi @FieldAppEngineerCombain and thanks for your contribution :) can you pls list your template into templates.yml file?

adriansmares commented on traxmate:
```diff
@@ -0,0 +1,22 @@
+    name: Servicetoken
+    description: Traxmate API Servicetoken
+    secret: true
+  - id: devEUI
```
This shouldn't be a template field, if it is really meant to identify the end device and not the integration.

See https://www.thethingsindustries.com/docs/integrations/webhooks/path-variables/ .

FieldAppEngineerCombain replied:
I have removed that and set it up as intended I think.

adriansmares commented on traxmate:
```diff
@@ -0,0 +1,22 @@
+headers:
+  Authorization: Token {token}
+create-downlink-api-key: true
+base-url: http://online.traxmate.io:3000/capture/{token}/device/{devEUI}
```
Do you have an `https` endpoint available ?

YllkaX replied:
Hi Adrian, 

yes the endpoint is : https://capture.v1.traxmate.io/capture/{token}/device/{devEUI}
Contributor

FieldAppEngineerCombain replied:
I have also updated the endpoint as YllkaX wanted

YllkaX commented:
Hi All!


Could you please help us out here? 



Regards, 

Yllka Shkupolli Xhemajli
Customer Success Manager 



￼



Combain Mobile AB
Ideon Science Park
Scheelevägen 27
223 70 Lund
Sweden
+46-(0)46-288 6 288


Mobile:  +46-(0)76 583 52 56
Email:  ***@***.***

NOTE: Interested in a meeting with me, book it here: https://calendly.com/yllkax
www.combain.com
www.traxmate.io



> 26 nov. 2023 kl. 19:37 skrev Nejra Selimović ***@***.***>:
> 
> 
> @nejraselimovic commented on this pull request.
> 
> In traxmate.yml <https://github.com/TheThingsNetwork/lorawan-webhook-templates/pull/90#discussion_r1405457551>:
> 
> > +description: Send data to Traxmate via TTI adapter
> +logo-url: https://traxmate.io/wp-content/uploads/2023/10/8429011b-92b9-4f53-b9fd-49285d29c437.png
> +info-url: https://traxmate.io
> +documentation-url: https://traxmate.io/support/getting-started/
> +fields:
> +  - id: token
> +    name: Servicetoken
> +    description: Traxmate API Servicetoken - found under Settings -> Account
> +    secret: true
> +format: json
> +headers:
> +  Authorization: Token {token}
> +create-downlink-api-key: true
> +base-url: https://capture.v1.traxmate.io/capture/{token}
> +paths:
> +  uplink-message: /device{/devID}
> hey @FieldAppEngineerCombain <https://github.com/FieldAppEngineerCombain>, sorry for the delay, last time I checked we needed to do a fix on our end to support this. @adriansmares <https://github.com/adriansmares> what's our status here?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/TheThingsNetwork/lorawan-webhook-templates/pull/90#discussion_r1405457551>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/BDW4NSXWFF7M66BS7BFEO4LYGOD5ZAVCNFSM6AAAAAA6ZRSYPWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTONBZGM4TOOJWGA>.
> You are receiving this because you commented.
>

adriansmares commented on traxmate.yml:
```diff
@@ -0,0 +1,19 @@
+create-downlink-api-key: true
+base-url: https://capture.v1.traxmate.io/capture/{token}
+paths:
+  uplink-message: /device{/devID}
```
Hi - my apologies for the long delay. I was wrong and the template variables like `devID` or `devEUI` cannot be currently used as part of webhook templates (only manually created webhooks).

Please use a fixed uplink path (like `/device`) and lookup the device ID in the request body - it will always be available under `end_device_ids.device_id` in the JSON.

FieldAppEngineerCombain replied:
Hmm okay, need to rethink how we do it then. I'll get back to you with a shortened path shortly.

FieldAppEngineerCombain commented:
Hi again,
I added back the format:json line and updated a header. If it fails again, is it possible to see more extensive error logs or do I need to debug it locally as suggested in the instructions? =)
@adriansmares
